zypp/repo/ServiceRepos.cc

```diff
@@ -40,10 +40,17 @@ namespace zypp
 	mediamanager.attach( mid );
 	mediamanager.provideFile( mid, "repo/repoindex.xml" );
 	Pathname path = mediamanager.localPath(mid, "repo/repoindex.xml" );
-	parser::RepoindexFileReader reader(path, callback);
-	service.setProbedTtl( reader.ttl() );	// hack! Modifying the const Service to set parsed TTL
-	mediamanager.release( mid );
-	mediamanager.close( mid );
+	try {
+	  parser::RepoindexFileReader reader(path, callback);
+	  service.setProbedTtl( reader.ttl() );	// hack! Modifying the const Service to set parsed TTL
+	  mediamanager.release( mid );
+	  mediamanager.close( mid );
+	} catch ( const Exception &e ) {
+	  //Reader throws a bare exception, we need to translate it into something our calling
+	  //code expects and handles (bnc#1116840)
+	  ZYPP_CAUGHT ( e );
+	  ZYPP_THROW( ServicePluginInformalException ( e.msg() ) );
```

Comment:
We should use [Exception::remember](https://github.com/openSUSE/libzypp/blob/master/zypp/base/Exception.h#L216) to store the original Exception as history.